File: Partition Labels/Partition Labels.py
# Solution for Partition Labels in PY

class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        n = len(s)
        mp = [-1] * 26
        for i in range(n):
            mp[ord(s[i]) - ord('a')] = i
        i, start, end = 0, 0, 0
        result = []
        while i < n:
            end = max(end, mp[ord(s[i]) - ord('a')])
            if i == end:
                result.append(end - start + 1)
                start = end + 1
            i += 1
        return result

File: Partition Labels/Partition Labels.cpp
// Solution for Partition Labels in CPP

class Solution {
public:
    vector<int> partitionLabels(string s) {
        int n = s.length();
        vector<int> mp(26, -1);
        for (int i = 0; i < n; i++) {
            mp[s[i] - 'a'] = i;
        }
        int i = 0, start = 0, end = 0;
        vector<int> result;
        while (i < n) {
            end = max(end, mp[s[i] - 'a']);
            if (i == end) {
                result.push_back(end - start + 1);
                start = end + 1;
            }
            i++;
        }
        return result;
    }
};
